README.md

```diff
@@ -1 +1,66 @@
 # snare
+
+`snare` is a simple GitHub webhooks runner. When a request comes in, it runs an
+arbitrary program for that repository informing it of the event -- that program
+can then perform whatever actions it wants.
+
+
+## Basic usage
+
+`snare` requires three command-line arguments to be specified:
+
+```
+Usage: snare -p <port> -r <repo scripts dir> -s <secret>
+```
+
+where:
+
+ * `<port>` is a port number (e.g. 4567)
+ * `<repo scripts dir>` is the directory where programs for each repository
+   targeted by the webhook are stored. For a repository `repo` owned by `user`
+   the command `<repo scripts dir>/<user>_<repo> <event> <path to GitHub JSON>`
+   will be run.
+ * `<secret>` is the GitHub secret which guarantees that hooks are coming from
+   your GitHub repository and not a malfeasant.
+
+The most important part of this are the per-repo programs. For example,
+`snare`'s GitHub repository is
+[`https://github.com/softdevteam/snare`](https://github.com/softdevteam/snare):
+if we set up a web hook up that notifies us of when pull request events have
+been made then the command `<repo scripts dir>/softdevteam_snare pull_request
+/path/to/json` will be executed where `pull_request` is the name of the GitHub
+event and `/path/to/JSON` is a path to a file containing the complete GitHub
```

Comment:
Is it `json` or `JSON`?

Reply:
Good spot. In the path it should be `json.